src/common/components/layouts/pages/inspect/Inspect.js

```diff
@@ -9,17 +9,20 @@ import { JOB_STATUS, TASK_STATUS } from '../../../../store/constants';
 import { WARNING_MESSAGES } from '../../../../services/constants';
 import { lockApp, resetOnFileUpload, unlockApp } from '../../../../store/application/actions';
 import { getJobStatus, getTaskStatus } from '../../../../store/job/selectors';
+import { getRuleSummaries } from '../../../../store/job/result/selectors';
 import Toolbar from './toolbar/Toolbar';
 import Tree from './tree/Tree';
 import PdfDocument from './pdfDocument/PdfDocument';
 import DropzoneWrapper from '../upload/dropzoneWrapper/DropzoneWrapper';
 
 import './Inspect.scss';
 
-function Inspect({ jobStatus, taskStatus, lockApp, unlockApp, onFileDrop }) {
+function Inspect({ jobStatus, taskStatus, ruleSummaries, lockApp, unlockApp, onFileDrop }) {
```

Comment:
Add a `useCallback` method in Inspect component, that will receive the index of the rule and toggle it's list (open/close it). Pass this method to other components and use whenever needed. For example:
```
    const onExpandRule = useCallback(
        (index, closeIfExists = true) => {
            const copyExpandedRule = _.clone(expandedRule);
            expandedRule.includes(index) && closeIfExists
                ? copyExpandedRule.splice(index, 1, UNSELECTED)
                : copyExpandedRule.splice(index, 1, index);
            return setExpandedRule(copyExpandedRule);
        },
        [expandedRule, setExpandedRule]
    );
```
PS: use lodash `_.clone` method instead of serialization-deserialization.
PPS: call this method as `onExpandRule(index)` if rule needs to be closed (e.g. on second click in the tree). call this method as `onExpandRule(index, false)` if rule should stay open even if it is already opened (e.g. when selecting the checks in one rule it should stay open)